File: src/BetterLyrics.KMP/shared/src/commonMain/kotlin/com/jayfunc/betterlyrics/interfaces/services/ILyricsCacheService.kt
package com.jayfunc.betterlyrics.interfaces.services
interface ILyricsCacheService {
    suspend fun initAsync()
    suspend fun getLyricsAsync(songInfo: Any, token: Any): Any?
    suspend fun saveLyricsAsync(songInfo: Any, result: Any, token: Any)
    suspend fun saveLyricsManualOverrideAsync(title: String, artist: String, album: String, raw: String, translation: String)
    suspend fun removeLyricsAsync(songInfo: Any)
    suspend fun clearAllLyricsAsync()
    suspend fun clearEmptyLyricsAsync()
}

File: src/BetterLyrics.DotNet/BetterLyrics.Core/Interfaces/Services/ILyricsCacheService.cs
using BetterLyrics.Core.Enums;
using BetterLyrics.Core.Models;
using BetterLyrics.Core.Models.Entities;

namespace BetterLyrics.Core.Interfaces.Services;

public interface ILyricsCacheService
{
    /// <summary>
    /// </summary>
    /// <param name="songInfo"></param>
    /// <param name="provider"></param>
    /// <param name="token"></param>
    /// <exception cref="OperationCanceledException"></exception>
    /// <returns></returns>
    Task<LyricsCacheItem?> GetLyricsAsync(SongInfo songInfo, LyricsProvider provider, CancellationToken token);

    /// <summary>
    ///     Write or update cache to DB
    /// </summary>
    /// <param name="songInfo"></param>
    /// <param name="result"></param>
    /// <param name="token"></param>
    /// <exception cref="OperationCanceledException"></exception>
    /// <returns></returns>
    Task SaveLyricsAsync(SongInfo songInfo, LyricsCacheItem result, CancellationToken token);

    Task ClearCacheAsync();
}